refl1d/probe.py

```diff
@@ -280,6 +280,9 @@ def simulate_data(self, theory, noise=None):
 
         if noise is None:
             pass # use existing noise
+        elif isinstance(noise, (list, tuple, numpy.ndarray)) and len(noise) == len(theory):
+            self.dR = numpy.asarray(noise)
+            self.dR[self.dR==0] = 1e-11
         else:
             self.dR = 0.01*noise*self.Ro
             self.dR[self.dR==0] = 1e-11
```

Comment:
The array length test is not useful.  If the length is wrong, then it just falls into the next branch where it is used unconditionally.

I prefer to avoid isinstance() tests.  Just use numpy.asarray() and let numpy sort out the types.
 
How about:
```
if noise is None:
    pass
else:
    self.dR = np.asarray(noise)
    if len(self.dR.shape) == 0:  # noise is a scalar
        self.dR = 0.01*noise*self.Ro
    self.dR[self.dR == 0] = 1e-11
```
    
```

Reply:
You don't mind that someone can pass an array of the wrong size?
I modified the code to use your version, and I added something similar to ```ProbeSet```.
Anywhere else that would be affected?